### ISSUE TITLE
OpenOCD compliance test suite not working anymore

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#2817]
It looks like the OpenOCD compliance test suite can't be used anymore with the newest version of OpenOCD, since the `-rtos riscv` option has been deprecated and replaced by `-rtos hwthreads`. If you try to run the OpenOCD compliance test suite as described in our documentation

https://docs.opentitan.org/doc/ug/getting_started_verilator/

you get the following error: `Error: Please run with -rtos riscv to run compliance test.`

See also https://github.com/riscv/riscv-openocd/issues/462#issuecomment-606138313 where this issue has been brought up as well. The answer was basically that the compliance test suite is not officially supported, and hence may be broken. 

I did some experiments and it looks like we just have to make sure that the `-rtos riscv` -> `-rtos hwthreads` change is also propagated to the parts of the code performing the compliance test, see patch here:

https://github.com/msfschaffner/riscv-openocd/commit/fc81737cf92c82b2ebfc2aee453debf565750cf0

With this change I can run the test suite again - but I haven't checked whether that breaks other things in OpenOCD.

We can raise this issue upstream, but if this can not be fixed (due to deprecation), we may have to update our documentation, or keep the patch in a lowRISC fork...

**Comment by GregAC** [comment]
Does sounds like they're not interested in the compliance test, but maybe they'd be happy to have a small PR to fix given it's still there.

Still probably not something we want to rely on for the future, bit of a shame really, perhaps we could provide our own smoke test  in OT that shows the debugger infrastructure is working?

**Comment by imphil** [comment]
The compliance test was never really supported (since it actually doesn't really test compliance), but it is still a very useful test which has uncovered a fair amount of issues in our integration. So if we can keep it working it would be helpful. As Greg said, upstream will most likely accept a patch to unbreak things, otherwise we can have it in our fork.

I can have a look at it some time if nobody beats me to it.

**Comment by msfschaffner** [comment]
Yes indeed, I have also found it a good test to run whenever you make modifications to the debug infrastructure - be it just a new vendor-in as in the linked PR.

I meant to make a PR upstream to ask whether that could be a solution, but haven't gotten around to that yet. So if anyone finds some time to look into that, that would of course be great ;).

**Comment by moidx** [comment]
CC @arunthomas @sriyerg 

Marking as M2 for further triaging.

**Comment by GregAC** [comment]
As I understand it the compliance test was never properly supported and has been removed/deprecated in later versions of OpenOCD. So we can close this.

**Comment by moidx** [comment]
Thanks @GregAC! Marking as not planned.
